lib/Extractor/Candidates.cpp

```diff
@@ -861,6 +865,71 @@ std::tuple<BlockPCs, std::vector<InstMapping>> GetRelevantPCs(
   return std::make_tuple(RelevantBPCs, RelevantPCs);
 }
 
+void PrintDataflowInfo(Function &F, Instruction &I, LazyValueInfo *LVI,
+                       ScalarEvolution *SE) {
+  auto V = I.getOperand(0);
+  auto DL = F.getParent()->getDataLayout();
+  if (PrintNegAtReturn) {
+    bool Negative = isKnownNegative(V, DL);
+    llvm::outs() << "negative from compiler: ";
+    if (Negative)
+      llvm::outs() << "true\n";
```

Comment:
ok this is getting there, thanks.

there's one more very easy way to simplify this code, please see here: http://www.cplusplus.com/reference/ios/boolalpha/

just set this flag at the start of PrintDataflowInfo() and clear it at the end

Reply:
Thanks for sharing this flag. Hope I have used it the right way.

Reply:
@regehr std::boolalpha won't work together with llvm::errs() and outs().